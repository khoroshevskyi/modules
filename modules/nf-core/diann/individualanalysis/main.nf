process DIANN_INDIVIDUALANALYSIS {
    tag "$ms_file.baseName"
    label 'process_high'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://containers.biocontainers.pro/s3/SingImgsRepo/diann/v1.8.1_cv1/diann_v1.8.1_cv1.img' :
        'docker.io/biocontainers/diann:v1.8.1_cv1' }"

    input:
    tuple val(meta), path(ms_file), path(fasta), path(library)

    output:
    tuple val(meta), path("*.quant"), emit: diann_quant
    tuple val(meta), path("*_final_diann.log"), emit: log
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "DIANN_INDIVIDUALANALYSIS module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    def args = task.ext.args ?: ''

    """
    diann   --lib ${library} \\
            --f ${ms_file} \\
            --fasta ${fasta} \\
            --threads ${task.cpus} \\
            --temp ./ \\
            ${args}

    cp report.log.txt ${ms_file.baseName}_final_diann.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        DIA-NN: \$(diann 2>&1 | grep "DIA-NN" | grep -oP "\\d+\\.\\d+(\\.\\w+)*(\\.[\\d]+)?")
    END_VERSIONS
    """

    stub:
    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        error "DIANN_INDIVIDUALANALYSIS module does not support Conda. Please use Docker / Singularity / Podman instead."
    }
    def args = task.ext.args ?: ''

    """
    touch ${ms_file.baseName}.quant
    touch ${ms_file.baseName}_final_diann.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        DIA-NN: \$(diann 2>&1 | grep "DIA-NN" | grep -oP "\\d+\\.\\d+(\\.\\w+)*(\\.[\\d]+)?")
    END_VERSIONS
    """
}
