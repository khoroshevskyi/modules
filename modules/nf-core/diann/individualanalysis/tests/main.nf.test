nextflow_process {

    name "Test Process DIANN_INDIVIDUALANALYSIS"
    script "../main.nf"
    process "DIANN_INDIVIDUALANALYSIS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "diann"
    tag "diann/individualanalysis"
    tag "untar"

    setup {
        run("UNTAR") {
            script "../../../../nf-core/untar/main.nf"
            process {
                """
                input[0] = [
                    [id:'test'],
                    file(params.modules_testdata_base_path + 'proteomics/diann/RD139_Narrow_UPS1_0_1fmol_inj1.mzML.tar.gz', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("proteomics - individual analysis") {

        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, dir ->  
                    def mzmlFile = dir.resolve('RD139_Narrow_UPS1_0_1fmol_inj1.mzML')
                    [meta, mzmlFile, file(params.modules_testdata_base_path + 'proteomics/diann/REF_EColi_K12_UPS1_combined_subset_100.fasta', checkIfExists: true), file(params.modules_testdata_base_path + 'proteomics/diann/empirical_library.speclib', checkIfExists: true)]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path("${process.out.log[0][1]}").exists() },
                { assert snapshot(
                    process.out.diann_quant,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("proteomics - individual analysis - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = UNTAR.out.untar.map{ meta, dir ->  
                    def mzmlFile = dir.resolve('RD139_Narrow_UPS1_0_1fmol_inj1.mzML')
                    [meta, mzmlFile, file(params.modules_testdata_base_path + 'proteomics/diann/REF_EColi_K12_UPS1_combined_subset_100.fasta', checkIfExists: true), file(params.modules_testdata_base_path + 'proteomics/diann/empirical_library.speclib', checkIfExists: true)]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
