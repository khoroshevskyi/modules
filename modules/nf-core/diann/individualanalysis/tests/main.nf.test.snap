{
    "proteomics - individual analysis": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "peakpicker_tutorial_1_mzML.quant:md5,20f8e7c64fcd264389b97799039c28de"
                ]
            ],
            [
                "versions.yml:md5,0fbc3a5aa136989fe74c1182e93f4776"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-24T14:44:05.043322"
    },
    "proteomics - individual analysis - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "peakpicker_tutorial_1.quant:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "peakpicker_tutorial_1_final_diann.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,0fbc3a5aa136989fe74c1182e93f4776"
                ],
                "diann_quant": [
                    [
                        {
                            "id": "test"
                        },
                        "peakpicker_tutorial_1.quant:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "peakpicker_tutorial_1_final_diann.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0fbc3a5aa136989fe74c1182e93f4776"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-24T14:44:10.02119"
    }
}